src/lua/lua_task.c

```diff
@@ -2644,22 +2691,22 @@ lua_task_inject_url(lua_State *L)
 
 	if (lua_isuserdata(L, 3)) {
 		/* We also have a mime part there */
-		mpart = *((struct rspamd_mime_part **) rspamd_lua_check_udata_maybe(L,
-																			3, rspamd_mimepart_classname));
+		mpart = *((struct rspamd_mime_part **)
+							   rspamd_lua_check_udata_maybe(L,3,rspamd_mimepart_classname));
 	}
-
 	if (task && task->message && url && url->url) {
 		if (rspamd_url_set_add_or_increase(MESSAGE_FIELD(task, urls), url->url, false)) {
-			if (mpart && mpart->urls) {
-				/* Also add url to the mime part */
-				g_ptr_array_add(mpart->urls, url->url);
+			if(mpart == NULL) {
```

Comment:
We don't need a fake mpart, why do we?

Reply:
So, we also do not need to extract mime part form Lua State?